dapr_agents/tool/http/client.py

```diff
@@ -0,0 +1,222 @@
+import os
+from typing import Dict, Literal, Optional, Any, Union
+from distutils.util import strtobool
+import logging
+import requests
+
+from pydantic import BaseModel, Field, PrivateAttr
+from dapr_agents.types import ToolError
+from urllib.parse import urlparse
+from opentelemetry.instrumentation.requests import RequestsInstrumentor
+from opentelemetry import trace
+from opentelemetry._logs import set_logger_provider
+
+
+logger = logging.getLogger(__name__)
+
+
+class DaprHTTPClient(BaseModel):
+    """
+    Client for sending HTTP requests to Dapr endpoints.
+    """
+
+    dapr_app_id: Optional[str] = Field(
+        default="", description="Optional name of the Dapr App ID to invoke."
+    )
+
+    dapr_http_endpoint: Optional[str] = Field(
+        default="",
+        description="Optional name of the HTTPEndpoint to call for invocation",
+    )
+
+    http_endpoint: Optional[str] = Field(
+        default="", description="Optional FQDN URL to request to."
+    )
+
+    method: Optional[str] = Field(
+        default="", description="Optional name of the method to invoke."
+    )
+
+    headers: Optional[Dict[str, str]] = Field(
+        default={},
+        description="Default headers to include in all requests.",
+    )
+
+    # Private attributes not exposed in model schema
+    _base_url: str = PrivateAttr(default="http://localhost:3500/v1.0/invoke")
+
+    def model_post_init(self, __context: Any) -> None:
+        """Initialize the client after the model is created."""
+
+        try:
+            otel_enabled: bool = bool(
+                strtobool(os.getenv("DAPR_AGENTS_OTEL_ENABLED", "True"))
+            )
+        except ValueError:
+            otel_enabled = False
+
+        if otel_enabled:
+            from dapr_agents.tool.utils.otel import DaprAgentOTel  # type: ignore[import-not-found]
+
+            otel_client = DaprAgentOTel(
+                service_name=os.getenv("OTEL_SERVICE_NAME", "dapr-http-client"),
+                otlp_endpoint=os.getenv("OTEL_EXPORTER_OTLP_ENDPOINT", ""),
+            )
+            tracer = otel_client.create_and_instrument_tracer_provider()
+            trace.set_tracer_provider(tracer)
+
+            otel_logger = otel_client.create_and_instrument_logging_provider(
+                logger=logger,
+            )
+            set_logger_provider(otel_logger)
+
+            RequestsInstrumentor().instrument()
+
+        logger.debug("Initializing DaprHTTPClient client")
+
+        super().model_post_init(__context)
+
+    def post(
```

Comment:
Would it be possible to have a single invoke method that accepts the HTTP verb as a parameter instead of individual methods for each verb?

Reply:
@yaron2 this is a good point. Have a look at fa680b3.

Should I also move the `otel.py` into a more proper location? I would assume under `dapr_agents.agent.utils`?

Reply:
I'd move it for the following import: `dapr_agents.agent.telemetry`

Reply:
agreed :)

Reply:
`otel.py` moved into `dapr_agents.agent.telemetry`.

Tested with the environment variable to enable/disable OTel traces from the http client.